trieste/space.py

```diff
@@ -91,6 +108,24 @@ def sample(self, num_samples: int) -> tf.Tensor:
 
         return tf.random.shuffle(self._points)[:num_samples, :]
 
+    def __mul__(self, dss: DiscreteSearchSpace) -> DiscreteSearchSpace:
+        """
+        Return the Cartesian product of the two :class:`DiscreteSearchSpace`
```

Comment:
```suggestion
        Return the Cartesian product of the two :class:`DiscreteSearchSpace`\ s. For example:
        
            >>> sa = DiscreteSearchSpace(tf.constant([[0, 1], [2, 3]]))
            >>> sb = DiscreteSearchSpace(tf.constant([[4, 5, 6], [7, 8, 9]]))
            >>> (sa * sb).points.numpy()
            array([[0, 1, 4, 5, 6],
                      [0, 1, 7, 8, 9],
                      [2, 3, 4, 5, 6],
                      [2, 3, 7, 8, 9]])

```
modulo some formatting